### PR TITLE
Added tests and resolved issue #42

### PR DESCRIPTION
**Pull request by InfoSec812**
Not sure WHY the Collectors.toMap() stream implementation is not working, but the simple case of using a `for` iterator seems to resolve the problem.

### CODE REVIEW

File: src/main/java/io/vertx/lang/groovy/ConversionHelper.java [hunk 1]
```diff
@@ -44,7 +45,7 @@ public static Object unwrap(Object obj) {
   }
 
   @SuppressWarnings("unchecked")
-  private static Object toJsonElement(Object obj) {
+  static Object toJsonElement(Object obj) {
```
Comment (InfoSec812) [inline src/main/java/io/vertx/lang/groovy/ConversionHelper.java]: Converted to package private for testing purposes.

File: src/main/java/io/vertx/lang/groovy/ConversionHelper.java [hunk 2]
```diff
@@ -58,9 +59,13 @@ private static Object toJsonElement(Object obj) {
   }
 
   public static JsonObject toJsonObject(Map<String, Object> obj) {
-    return obj == null ? null : new JsonObject(obj.entrySet().stream().collect(Collectors.toMap(
-      Map.Entry::getKey, entry -> toJsonElement(entry.getValue())
```
Comment (InfoSec812) [inline src/main/java/io/vertx/lang/groovy/ConversionHelper.java]: This is point in the code where the NPE is being thrown, and so far I have been unable to determine why.

**Comment by vietj** [comment]
Fixed by https://github.com/vert-x3/vertx-lang-groovy/commit/f9c75a1a01fbb63a6850c560feb885f416637da3

I've created a vertx gen test interface that accept only a JsonObject with a single entry whose value is null.

**Comment by InfoSec812** [comment]
Woot!

**Pull request closed without merging**